### ISSUE TITLE
[Accessibility]: Search contexts management mobile UI issues

### ISSUE THREAD

**Issue by limitedmage** [sourcegraph/sourcegraph#34730]
### Audit type

Mobile device navigation

### User journey audit issue

https://github.com/sourcegraph/sourcegraph/issues/34191

### Problem description

The search context pages (list page, view page, and edit/create page) have significant mobile UI issues. It looks like a narrow screen/mobile UI was not designed or implemented for this page. 

List page:
![image](https://user-images.githubusercontent.com/206864/166076949-e7e520d6-d795-40bb-afaa-ffb780ff46d8.png)

View page:
![image](https://user-images.githubusercontent.com/206864/166076982-a0241653-872b-43cc-ba46-9ba43068a263.png)

Edit page:
![image](https://user-images.githubusercontent.com/206864/166076993-c7d3eba6-9794-48c6-b49e-f0d6a344934d.png)

### Expected behavior

The pages look good and behave correctly on mobile

### Assigning labels

- [X] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [X] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [X] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [ ] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

This issue will be fixed by my team, I have assigned a relevant member to it, or I will do so in the near future.

**Comment by sourcegraph-bot-2** [comment]
Heads up @benvenker @lguychard - the "team/search-product" label was applied to this issue.
